multiqc/modules/verifybamid/verifybamid.py

```diff
@@ -0,0 +1,238 @@
+                    # create a dictionary entry with the first column as a key (sample name) and an empty dictionary as a value
+                    parsed_data[s_name] = {}
+                    # for each item in list of items in the row
+                    for i, v in enumerate(s):
+                    # if it's not the first element (if it's not the name)
+                        if i != 0:
+                            # try and convert the value into a float
+                            try:
+                            # and add to the dictionary the key as the corrsponding item from the header and the value from the list
+                                parsed_data[s_name][headers[i]] = float(v)
+                                #if can't convert to float...
+                            except ValueError:
+                            # add to the dictionary the key as the corrsponding item from the header and the value from the list
+                                parsed_data[s_name][headers[i]] = v
+    
+        # nothing has been parsed return None
+        if len(parsed_data) == 0:
+            return None
+        # else return the dictionary
+        return parsed_data
+
+    def verifybamid_general_stats_table(self):
+        """ Take the percentage of contamination from all the parsed *.SELFSM files and add it to the basic stats table at the top of the report """
+
+        # create a dictionary to hold the columns to add to the general stats table
+        headers = OrderedDict()
+        # available columns are:
+        #SEQ_ID RG  CHIP_ID #SNPS   #READS  AVG_DP  FREEMIX FREELK1 FREELK0 FREE_RH FREE_RA CHIPMIX CHIPLK1 CHIPLK0 CHIP_RH CHIP_RA DPREF   RDPHET  RDPALT
+        #,see https://genome.sph.umich.edu/wiki/VerifyBamID#Interpreting_output_files
+
+        # add the CHIPMIX column. set the title and description
+        headers['CHIPMIX'] = {
+            'title': 'Contamination Prediction - sequence and array',
+            'description': 'VerifyBamID: CHIPMIX -   Sequence+array estimate of contamination (NA if the external genotype is unavailable) (0-1 scale)',
+            'format': '{:,.5f}' 
+        }
+
+        # add the FREEMIX column. set the title and description
+        headers['FREEMIX'] = {
+            'title': 'Contamination Prediction - sequence only',
+            'description': 'VerifyBamID: FREEMIX -   Sequence-only estimate of contamination (0-1 scale).',
+            'format': '{:,.5f}' 
+        }
+        
+        # pass the data dictionary and header dictionary to function to add to table.
+        self.general_stats_addcols(self.verifybamid_data, headers)
+
+        
+    def verifybamid_table(self):
+        """
+        Create a table with all the columns from verify BAM ID
+        """
+        # create empty dictionary
+        data = dict()
+        # for each smaple name and the corresponding dictionary
+        for s_name, d in self.verifybamid_data.items():
```

Comment:
I'm not sure that you need to do this? It seems as though you're just copying `self.verifybamid_data`, in which case you can just use `self.verifybamid_data` directly...